core/primitives/src/block_header.rs

```diff
@@ -323,9 +325,9 @@ impl BlockHeaderV3 {
 #[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
 #[derive(BorshSerialize, BorshDeserialize, Serialize, Debug, Clone, Eq, PartialEq)]
 pub enum BlockHeader {
-    BlockHeaderV1(Box<BlockHeaderV1>),
-    BlockHeaderV2(Box<BlockHeaderV2>),
-    BlockHeaderV3(Box<BlockHeaderV3>),
+    BlockHeaderV1(Arc<BlockHeaderV1>),
```

Comment:
checking -- does Arc and Box serialize to the same thing ?

Reply:
Yes!